feature-libs/checkout/base/occ/adapters/occ-checkout-billing-address.adapter.spec.ts

```diff
@@ -0,0 +1,161 @@
+import { HttpClient, HttpErrorResponse } from '@angular/common/http';
+import {
+  HttpClientTestingModule,
+  HttpTestingController,
+} from '@angular/common/http/testing';
+import { fakeAsync, TestBed, tick } from '@angular/core/testing';
+import { Cart } from '@spartacus/cart/base/root';
+import {
+  Address,
+  ConverterService,
+  HttpErrorModel,
+  normalizeHttpError,
+  OccConfig,
+  OccEndpoints,
+} from '@spartacus/core';
+import { defer, of, throwError } from 'rxjs';
+import { take } from 'rxjs/operators';
+import { OccCheckoutBillingAddressAdapter } from './occ-checkout-billing-address.adapter';
+
+const userId = '123';
+const cartId = '456';
+const cartData: Partial<Cart> = {
+  store: 'electronics',
+  guid: '1212121',
+};
+
+const MockOccModuleConfig: OccConfig = {
+  backend: {
+    occ: {
+      baseUrl: '',
+      prefix: '',
+      endpoints: {
+        setBillingAddress:
+          'orgUsers/${userId}/carts/${cartId}/addresses/billing',
+      } as OccEndpoints,
+    },
+  },
+  context: {
+    baseSite: [''],
+  },
+};
+
+const mockJaloError = new HttpErrorResponse({
+  error: {
+    errors: [
+      {
+        message: 'The application has encountered an error',
+        type: 'JaloObjectNoLongerValidError',
+      },
+    ],
+  },
+});
+const mockNormalizedJaloError = normalizeHttpError(mockJaloError);
+
+describe(`OccCheckoutBillingAddressAdapter`, () => {
+  let service: OccCheckoutBillingAddressAdapter;
+  let httpClient: HttpClient;
+  let httpMock: HttpTestingController;
+  let converter: ConverterService;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      imports: [HttpClientTestingModule],
+      providers: [
+        OccCheckoutBillingAddressAdapter,
+        { provide: OccConfig, useValue: MockOccModuleConfig },
+      ],
+    });
+    service = TestBed.inject(OccCheckoutBillingAddressAdapter);
+    httpClient = TestBed.inject(HttpClient);
+    httpMock = TestBed.inject(HttpTestingController);
+    converter = TestBed.inject(ConverterService);
+
+    spyOn(converter, 'pipeable').and.callThrough();
+    spyOn(converter, 'pipeableMany').and.callThrough();
+    spyOn(converter, 'convert').and.callThrough();
+  });
+
+  afterEach(() => {
+    httpMock.verify();
+  });
+
+  describe(`setAddress`, () => {
+    const address: Address = { country: 'Poland' } as Address;
+
+    it(`should set address for cart for given user id, cart id and address id`, (done) => {
+      service
+        .setAddress(userId, cartId, address)
+        .pipe(take(1))
+        .subscribe((result) => {
+          expect(result).toEqual(cartData);
+          done();
+        });
+
+      const mockReq = httpMock.expectOne((req) => {
+        return (
+          req.method === 'PUT' &&
+          req.url === `orgUsers/${userId}/carts/${cartId}/addresses/billing`
```

Comment:
orgUsers? Iam not sure why we compare to B2B endpoint here.

Reply:
changed